Adding recommended checklist for staging review

1Copenut opened the pull request:
We are looking to swap out the link to our (deprecated) Markdown file and include the required and recommended checklists in the VSP collaboration cycle documents.

1Copenut committed:
Adding recommended checklist for staging review

We are looking to swap out the link to our (deprecated) Markdown file and include the required and recommended checklists in the VSP collaboration cycle documents.

1Copenut committed:
Update staging-review-processes.md

1Copenut committed:
Update vsp-collaboration-cycle.md

jenstrickland commented on platform/working-with-vsp/vsp-collaboration-cycle/vsp-collaboration-cycle.md:
```diff
@@ -313,11 +313,21 @@ You bring:
 - Link to your [test plan](https://dsvavsp.testrail.io/index.php?/suites/view/2&group_by=cases:section_id&group_order=asc) in TestRail with [test results](https://dsvavsp.testrail.io/index.php?/runs/view/7&group_by=cases:section_id&group_order=asc) _for QA practice area review_ 
```
Might we clarify this to `Link to your [QA test plan]`?

1Copenut replied:
👍 Done.

jenstrickland commented on platform/working-with-vsp/vsp-collaboration-cycle/vsp-collaboration-cycle.md:
```diff
@@ -313,11 +313,21 @@ You bring:
 - Link to your [test plan](https://dsvavsp.testrail.io/index.php?/suites/view/2&group_by=cases:section_id&group_order=asc) in TestRail with [test results](https://dsvavsp.testrail.io/index.php?/runs/view/7&group_by=cases:section_id&group_order=asc) _for QA practice area review_ 
 - Link to your [Coverage for References](https://dsvavsp.testrail.io/index.php?/reports/view/12) and [Summary(Defects)](https://dsvavsp.testrail.io/index.php?/reports/view/14) reports in TestRail _for QA practice area review_
```
Might we be specific here, too? I think this is QA again.

1Copenut replied:
The QA practice area review is defined at the end of this sentence. I'm going to opt to leave this one as is.

jenstrickland commented on platform/working-with-vsp/vsp-collaboration-cycle/vsp-collaboration-cycle.md:
```diff
@@ -313,11 +313,21 @@ You bring:
-  - Link to a doc in your product folder that indicates all required checks from the accessibility [staging review prep](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/staging-review-processes.md) doc are complete.
+- Verify low-level accessibility tests are complete. Choose one of these 2 options:
+  - **(preferred)** Use the [TestRail VSP low-level accessibility testing checklist](https://dsvavsp.testrail.io/index.php?/suites/view/14&group_by=cases:section_id&group_order=asc).
+  <br/>**OR**
```
What is the benefit of creating a separate doc in the team's product folder, rather than having them certify the checklist in the request ticket? It's easier for the teams and the a11y specialist if the checklist is in the staging review request ticket, imho.

1Copenut replied:
@jenstrickland That's a good point. So instead of copy/pasting it into a new doc, we'd change that line to ask teams to check off the list in the staging review or full audit issue they open.

@gmaciolek What do you think about this? I could go with that, letting PMs fill out the checklist in issue, instead of having to create a new doc, and link to it.

gmaciolek replied:
I'm fine with that! I don't even recall how we arrived at linking to a doc in the product folder. I think it's probably a carry-over from when we were asking people to create a Markdown file using a template? But asking teams to certify the checks are complete in the ticket seems like a much better solution to me.

1Copenut replied:
Thank you both. I'll make these changes after a few meetings.

1Copenut committed:
Update vsp-collaboration-cycle.md

Clarifying a link to QA test plan

1Copenut committed:
Update vsp-collaboration-cycle.md

1Copenut committed:
Update vsp-collaboration-cycle.md

Updating language around the staging review and audit checklists.

1Copenut commented:
@mandifrances && @andreahewitt-odd Ready for re-review. I'll update the staging review and 508 audit templates with the checklist when this is approved and merged, for consistency.

jenstrickland commented on platform/accessibility/guidance/staging-review-processes.md:
```diff
@@ -9,20 +9,26 @@ Staging reviews test the [happy path](https://en.wikipedia.org/wiki/Happy_path)
 ## Required accessibility checklist for staging review
```
In the second paragraph of the Accessibility staging review preparation, there is this line:

> Accessibility specialists conduct a [more thorough pre-launch audit](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/508-request-prelaunch-review.md) when an application is relatively code-stable in production.)

We no longer conduct this Pre-launch 508 Review, and it was replaced with the Staging Review (code stable in staging) and Full A11y & 508 Office Audit (after launch).

I believe the Pre-launch 508 Review was more effective for catching accessibility issues, reducing the number found post-launch. The 2-day turnaround requirement of the staging review led to removing this.

1Copenut replied:
Updated the URL and changed this to post-launch language

jenstrickland commented on platform/accessibility/guidance/staging-review-processes.md:
```diff
@@ -9,20 +9,26 @@ Staging reviews test the [happy path](https://en.wikipedia.org/wiki/Happy_path)
-- [ ] [Screen reader tests](#screen-reader-tests)
+- [ ] [Screen reader tests (VoiceOver + Safari or NVDA + Firefox)](#screen-reader-tests)
+
+## Recommended accessibility checklist for staging review
```
Consider a title of "Recommended tests"

jenstrickland commented on platform/accessibility/guidance/staging-review-processes.md:
```diff
@@ -9,20 +9,26 @@ Staging reviews test the [happy path](https://en.wikipedia.org/wiki/Happy_path)
 ## Required accessibility checklist for staging review
```
Consider a title of "Required tests"

jenstrickland commented on platform/accessibility/guidance/staging-review-processes.md:
```diff
@@ -9,20 +9,26 @@ Staging reviews test the [happy path](https://en.wikipedia.org/wiki/Happy_path)
+    * **(preferred)** Use the [TestRail VSP low-level accessibility testing checklist](https://dsvavsp.testrail.io/index.php?/projects/overview/13). The accessibility test cases template is available in TestRail for you to copy into your test plan and includes all the required accessibility checks.   \
 OR
-    *   Copy the checklist from this doc into a doc in your product folder. _Note that this artifact can be used for the Collaboration Cycle staging review._
+    *   Copy the required and recommended checklists from this doc into a doc in your product folder. _Note that this artifact can be used for the Collaboration Cycle staging review._
```
Consider removing this option, and instead include the checklist of requirements in the staging review request — which may include more checklists for other practice areas. I'd like to minimize effort for VFS teams, as they're already having a tough time incorporating this into their workflows.

jenstrickland replied:
In the below "How to request help" section, it asks teams to create 508 issue tickets for questions. I'm finding teams have questions as they're doing their "low-level" tests. Might we 1) supplement guidance provided to include info that if they see issues in the header, footer, or with the Feedback button to disregard; how to handle background color manual checks, etc, and 2) revise to say to open 508 issues for issues discovered, rather than to use for questions?

jenstrickland replied:
For each of the "low-level" tests teams need guidance on what to disregard, as they're currently asking lots of questions and including screenshots of axe scan reports, etc.

jenstrickland replied:
Below it says

> After you request a staging review, the accessibility specialist reviews your work and documents any defects in ZenHub tickets. Issues with a 508 defect rating of “0” or “1” are launch blockers. Others can be fixed post-launch.  

Might we be specific that 508-defect-2 and 508-defect-3 issues **must** be fixed within the next [insert number] sprints?

For 508-defect-4 issues, consider clarification these are enhancements, and decisions should be documented within [insert number] sprints. 

This may help resolve the backlog of accessibility issues.

1Copenut committed:
Update staging-review-processes.md

Updating low-level to foundational language

1Copenut committed:
Update staging-review-processes.md

Updating language and listings for the required and recommended accessibility tests

1Copenut committed:
Update staging-review-processes.md

Updated language around the post-launch audit and URL

1Copenut committed:
Update staging-review-processes.md

Adding language and timeline for fixing issues

1Copenut committed:
Update staging-review-processes.md

Making the checklist into simple bulleted lists. The checklist doesn't make as much sense here as it will on the actual request tickets.

jenstrickland commented on platform/accessibility/guidance/staging-review-processes.md:
```diff
@@ -3,26 +3,32 @@
+    * **(preferred)** Use the [TestRail VSP foundational accessibility testing checklist](https://dsvavsp.testrail.io/index.php?/projects/overview/13). The accessibility test cases template is available in TestRail for you to copy into your test plan and includes all the required accessibility checks.   \
 OR
-    *   Copy the checklist from this doc into a doc in your product folder. _Note that this artifact can be used for the Collaboration Cycle staging review._
+    *   Copy the required and recommended checklists from this doc into a doc in your product folder. _Note that this artifact can be used for the Collaboration Cycle staging review._
```
Ruh-roh… this is what the checkboxes were for. 🤦‍♀️  Can we just put a checklist in the request ticket? Maybe it's commented out, and the guidance here tells them to uncomment it?

jenstrickland commented on platform/accessibility/guidance/staging-review-processes.md:
```diff
@@ -3,26 +3,32 @@
-    *   Copy the checklist from this doc into a doc in your product folder. _Note that this artifact can be used for the Collaboration Cycle staging review._
+    *   Copy the required and recommended checklists from this doc into a doc in your product folder. _Note that this artifact can be used for the Collaboration Cycle staging review._
 
 2. Link to any unresolved accessibility ZenHub issues in your staging review request ticket.
```
When they link to any unresolved issues (I don't think we need to say Zenhub), can we provide an example? When they provide a link to their staging a11y review feedback instead of the linked list of individual issues I have been creating the list myself, so that I have it for the 508 Office.

jenstrickland replied:
For the section below on how to request help, I don't understand the first bullet. I think it says if they have specific questions about issues from accessibility checks to open a 508 issue. Instead, I think we want to say, if you find issues during your team's own accessibility tests, open a 508 issue and then if not resolved… the next bullet.

jenstrickland commented on platform/accessibility/guidance/staging-review-processes.md:
```diff
@@ -37,9 +43,15 @@ OR
 ## What to expect from accessibility specialists during staging review
 
-After you request a staging review, the accessibility specialist reviews your work and documents any defects in ZenHub tickets. Issues with a 508 defect rating of “0” or “1” are launch blockers. Others can be fixed post-launch.  
+After you request a staging review, the accessibility specialist reviews your work and documents any defects in ZenHub tickets. **Issues with a 508 defect rating of “0” or “1” are launch blockers.** Issues with severity raings of "2, 3, or 4" should be fixed according to the following schedule:
```
Consider moving the `**Issues with a 508 defect rating of “0” or “1” are launch blockers.** ` to be the first bullet, and then follow with the others. I'm worried it will get missed inside the paragraph. 

Also, consider bolding the when they should be fixed.

jenstrickland commented on platform/accessibility/guidance/staging-review-processes.md:
```diff
@@ -37,9 +43,15 @@ OR
+* **508-defect-3** issues should be fixed in 1-3 sprints
+* **508-defect-4** issues should be considered for fixes or exploration in 2-4 sprints
+
+The accessibility specialist attaches the tickets to a single ZenHub epic and assigns the epic to the person who requested the staging review. During the staging review meeting, you and your team can ask questions about any of the issues identified. Applications are not considered Section 508 compliant until all severity 0, 1, 2, and 3 issues are remediated. Severity 4 issues are treated as enhan ...
```
Severity 4 issues may also be minor issues, so consider removing the last sentence?

1Copenut committed:
Making some final changes ahead of merging.

1Copenut committed:
Updating a single use of the word ZenHub
